### PR TITLE
feat(DEV-2154): implement lazy contract methods

### PR DESCRIPTION
**Pull request by sneljo1**
- implement lazy contract methods on new contract classes
- move mintoffer logic to new lazy contract class
- fixed signTypedData not working on browser

So, basically we're now able to do something like this to access contract methods. We'll slowly introduce it for other contracts as we implement new features. As it is a good way to pull some of the core logic out of common area's
```ts
const lazymintContract = await refinableSDK.evm.getErc721LazyMintContract(
        <contractAddress>
 );

await lazymintContract.updateSaleSettings({
  maxPerMint: values.maxItemsPerTx,
  maxPerWallet: values.maxItemsPerWallet,
});

```

### CODE REVIEW

File: src/nft/ERC1155NFT.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class ERC1155NFT extends AbstractEvmNFT {
   }
 
   async approve(operatorAddress: string): Promise<EvmTransaction> {
-    const nftTokenContract = await this.getTokenContract();
+    const nftTokenContract = await this.getTokenContractWrapper();
```
Comment (maximgeerinck) [inline src/nft/ERC1155NFT.ts:26]: what's a contract wrapper. find this confusing tbh

File: tests/unit/utils/parse-error.spec.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ describe("Error handling", () => {
           accessList: null,
         },
       },
-      "Process failed. Please try again.",
+      "Unable to finalise this sale, it might no longer be active.",
```
Comment (maximgeerinck) [inline tests/unit/utils/parse-error.spec.ts:92]: `finalize`

**Review by maximgeerinck — APPROVED** [review]
Looking good, just confused about the wrapper thing

File: src/nft/ERC1155NFT.ts (hunk 1 above)
Comment (sneljo1) [reply src/nft/ERC1155NFT.ts:26]: Contract class doesn't do much by itself. Its a representation of the db model. Contractwrapper is a class that wraps around the ethers contract to actually invoke the contract with and do error handling, better gas estimation,....

Comment (sneljo1) [reply src/nft/ERC1155NFT.ts:26]: As you are seeing in this PR, I am still fine-tuning the usage of contractWrapper as we go. Not 100% happy yet with the API

**Pull request merged**